aquacomputer_d5next.c

```diff
@@ -476,12 +493,25 @@ static int aqc_percent_to_pwm(u16 val)
 /* Converts to centi-percent */
 static int aqc_pwm_to_percent(long val)
 {
-	if (val < 0 || val > 255)
```

Comment:
Don't remove this and clamp later. Sysfs specs say

```
pwm[1-*]	Pulse width modulation fan control.
		Integer value in the range 0 to 255
		RW
		255 is max or 100%.
```

and we have to follow that. If it's outside of [0, 255], it's an -EINVAL.

Reply:
I found this at https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface which suggests that clamping is allowed here.
> What to do if a value is found to be invalid, depends on the type of the
sysfs attribute that is being set. If it is a continuous setting like a
tempX_max or inX_max attribute, then the value should be clamped to its
limits using clamp_val(value, min_limit, max_limit). If it is not continuous
like for example a tempX_type, then when an invalid value is written,
-EINVAL should be returned.
>
> Example1, temp1_max, register is a signed 8 bit value (-128 - 127 degrees):

	long v = simple_strtol(buf, NULL, 10) / 1000;
	v = clamp_val(v, -128, 127);
	/* write v to register */

Reply:
Copying my comment from the thread, since gmail didn't route it here:

I'd argue that PWM doesn't fall into that, because it's well defined.
Besides, it was Guenter himself who requested that the check be added to
that function, and the code is in mainline, so let's make it as it was.